front-end/src/visitor/VisitorFunction.cpp

```diff
@@ -83,6 +170,28 @@ antlrcpp::Any StartVisitor::visitTypeList(MapleGrammarParser::TypeListContext *c
     return fParams;
 }
 
+antlrcpp::Any StartVisitor::visitTypeListWithoutName(MapleGrammarParser::TypeListWithoutNameContext *ctx) {
+    auto fParams = new vector<FunctionParam *>(ctx->ID().size());
+
+    for (std::size_t i = 0; i != ctx->ID().size(); i++) {
+        const string &name = ctx->ID(i)->getText();
+
+        if (auto symbol = currentSymbolTable->lookup(name)) {
+            cerr << "Duplicate declaration of " << name << endl;
+            cerr << "Found : " << symbol->getDeclaration() << endl;
+            printDebugInfo(cerr, ctx);
+            throw std::runtime_error("Duplicated declaration");
+        }
```

Comment:
Like I said, no need to check for the name now.
Just create the vector of functionParam and it will be checked in `visitFunctionDeclaration`